### DISCUSSION TITLE
SLSA Github Generator

### DISCUSSION THREAD

**Discussion by gotson** [jreleaser/jreleaser#1334] [General]
Hello,

while looking at SLSA (i received a PR about it in one repo i maintain), i noticed that JReleaser does publish SLSA provenance.

Would it make sense to add some documentation on how to support SLSA Provenance using JReleaser somewhere ? Maybe there should be a JReleaser section in https://github.com/slsa-framework/slsa-github-generator/blob/main/internal/builders/generic/README.md#integration-with-other-build-systems (there is one for GoReleaser already), or maybe within the JReleaser website ?

**Comment by gotson** [comment]
side question @aalmiray, i am wondering why you are using [your own provenance uploader](https://github.com/jreleaser/jreleaser/blob/main/.github/scripts/upload-provenance.sh) instead of using `upload-tag-name` of the workflow: https://github.com/slsa-framework/slsa-github-generator/blob/main/internal/builders/generic/README.md#workflow-inputs

**Reply by aalmiray** [reply]
Last time I used their action it would create a new release instead of attaching the asset to the freshly created release. I have to try it out once more and see if I can remove my custom uploader.

**Reply by aalmiray** [reply]
Yeah... no dice when the release is created before provenance. At first I thought it had something to do with prerelease but no:

- early-access https://github.com/aalmiray/foobar/actions/runs/4719106429/jobs/8369550938
- release https://github.com/aalmiray/foobar/actions/runs/4719277154/jobs/8369941806

https://github.com/aalmiray/foobar/blob/fd249461aef587152e58564080231ac33bfe47f6/.github/workflows/release.yml#L88-L95

**Reply by aalmiray** [reply]
**UPDATE:** Just noticed I missed setting `upload-tag-name` in the provenance action. I think this makes it work 🎉

**Reply by aalmiray** [reply]
Sent a PR https://github.com/slsa-framework/slsa-github-generator/pull/2001

**Reply by aalmiray** [reply]
PR has been merged https://github.com/slsa-framework/slsa-github-generator/tree/main/internal/builders/generic#provenance-for-jreleaser

Next step: add a link to this resource at https://jreleaser.org
